src/Microsoft.AspNet.BasicMiddleware/OverrideHeaderMiddleware.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    public class OverrideHeaderMiddleware
+    {
+        private const string XForwardedFor = "X-Forwarded-For";
+        private const string XForwardedHost = "X-Forwarded-Host";
+        private const string XForwardedProto = "X-Forwarded-Proto";
+        private readonly OverrideHeaderMiddlewareOptions _options;
+        private readonly RequestDelegate _next;
+
+        public OverrideHeaderMiddleware(RequestDelegate next, OverrideHeaderMiddlewareOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(_next));
+            }
+            _options = options;
+            _next = next;
+        }
+
+        /// <summary>
```

Comment:
The `Invoke` method doesn't really need doc comments because users never call this method. On the other hand, lots of other types/members in this PR need doc comments (e.g. the `IApplicationBuilder` extension methods).